src/test/java/co/rsk/federate/signing/hsm/message/ReleaseCreationInformationGetterTest.java

```diff
@@ -348,4 +390,78 @@ void getTxInfoToSign_whenTransactionReceiptNotFoundInSubsequentBlock_shouldThrow
+    )
+        throws HSMReleaseCreationInformationException {
+        // Arrange
+        List<LogInfo> logs = new ArrayList<>();
+
+        CallTransaction.Function releaseRequestedEvent = BridgeEvents.RELEASE_REQUESTED.getEvent();
+        byte[] releaseRequestedSignatureTopic = releaseRequestedEvent.encodeSignatureLong();
+        List<DataWord> topics = new ArrayList<>();
+        topics.add(DataWord.valueOf(releaseRequestedSignatureTopic));
+        topics.add(DataWord.valueOf(pegoutCreationRskTx.getHash().getBytes()));
+        topics.add(DataWord.valueOf(pegoutBtcTx.getHash().getBytes()));
+        logs.add(new LogInfo(PrecompiledContracts.BRIDGE_ADDR.getBytes(), topics, null));
+
+        CallTransaction.Function pegoutTransactionCreatedEvent = BridgeEvents.PEGOUT_TRANSACTION_CREATED.getEvent();
+        byte[] pegoutTransactionCreatedSignatureTopic = pegoutTransactionCreatedEvent.encodeSignatureLong();
+        List<DataWord> pegoutTransactionCreatedTopics = new ArrayList<>();
+        pegoutTransactionCreatedTopics.add(DataWord.valueOf(pegoutTransactionCreatedSignatureTopic));
+        pegoutTransactionCreatedTopics.add(DataWord.valueOf(pegoutBtcTx.getHash().getBytes()));
+
+        byte[] pegoutTransactionCreatedEncodedData = pegoutTransactionCreatedEvent.encodeEventData(serializedOutpointValues);
+        logs.add(new LogInfo(PrecompiledContracts.BRIDGE_ADDR.getBytes(), pegoutTransactionCreatedTopics, pegoutTransactionCreatedEncodedData));
+
+        pegoutCreationRskTxReceipt.setLogInfoList(logs);
+
+        BlockStore blockStore = mock(BlockStore.class);
+        when(blockStore.getBlockByHash(pegoutCreationBlock.getHash().getBytes())).thenReturn(pegoutCreationBlock);
+        when(blockStore.getChainBlockByNumber(pegoutCreationBlock.getNumber())).thenReturn(pegoutCreationBlock);
+
+        ReceiptStore receiptStore = mock(ReceiptStore.class);
+        when(receiptStore.getInMainChain(pegoutCreationRskTx.getHash().getBytes(), blockStore)).thenReturn(Optional.of(
+            pegoutCreationRskTxInfo));
+
+        ReleaseCreationInformationGetter releaseCreationInformationGetter = new ReleaseCreationInformationGetter(
+            receiptStore,
+            blockStore
+        );
+
+        // act
+        ReleaseCreationInformation pegoutCreationInfo = releaseCreationInformationGetter.getTxInfoToSign(
+            version,
+            pegoutCreationRskTx.getHash(),
+            pegoutBtcTx
+        );
+
+        // assert
+        assertEquals(pegoutCreationBlock, pegoutCreationInfo.getPegoutCreationBlock());
+        assertEquals(pegoutCreationRskTxReceipt, pegoutCreationInfo.getTransactionReceipt());
+        assertEquals(pegoutCreationRskTx.getHash(), pegoutCreationInfo.getPegoutCreationRskTxHash());
+        assertEquals(pegoutBtcTx, pegoutCreationInfo.getPegoutBtcTx());
+        assertArrayEquals(expectedOutpointValues.toArray(), pegoutCreationInfo.getUtxoOutpointValues().toArray());
+    }
+
```

Comment:
What do you think about adding a test for when the `transactionReceipt` doesn't have the `pegout_transaction_created` the `utxoOutpointValues` list should be empty? Or update an existing test to include that assertion.
Let me know.

Reply:
There is already a test for that. Check the existing ok test.